go-app-ussd_sassa_registration.js

```diff
@@ -0,0 +1,203 @@
+            return "state_id_number";
+          }
+        },
+      });
+    });
+
+    self.states.add("state_exit", function (name) {
+      return new EndState(name, {
+        next: "state_start",
+        text: $(
+          "You must be a South African Citizen, Permanent Resident or Refugee to apply for this service."
+        ),
+      });
+    });
+
+    self.add("state_id_number", function (name, opts) {
+      var question = "Please enter your ID Number (eg 1234567890088)";
+      var error_msg = [
+        "Sorry, that is not a valid ID Number.",
+        "",
+        "Please enter your ID Number (eg 1234567890088)",
+      ].join("\n");
+
+      if (self.im.user.answers.state_start === "refugee") {
+        question =
+          "Please enter your Refugee Permit Number (e.g. 1234567890268)";
+        error_msg = [
+          "Sorry, that is not a valid Refugee Permit number.",
+          "",
+          "Please enter your Refugee Permit Number (eg 1234567890268)",
+        ].join("\n");
+      }
+
+      return new FreeText(name, {
+        question: question,
+        check: function (content) {
+          var match = content.match(/^(\d{6})(\d{4})(0|1)8\d$/);
+          var today = new moment(self.im.config.testing_today).startOf("day"),
+            dob;
+          var validLuhn = function (content) {
+            return (
+              content
+                .split("")
+                .reverse()
+                .reduce(function (sum, digit, i) {
+                  return (
+                    sum +
+                    _.parseInt(
+                      i % 2 ? [0, 2, 4, 6, 8, 1, 3, 5, 7, 9][digit] : digit
+                    )
+                  );
+                }, 0) %
+                10 ==
+              0
+            );
+          };
+          if (
+            !match ||
+            !validLuhn(content) ||
+            !(dob = new moment(match[1], "YYMMDD")) ||
+            !dob.isValid() ||
+            !dob.isBetween(
+              today.clone().add(-130, "years"),
+              today.clone().add(-17, "years")
+            )
+          ) {
+            return error_msg;
+          } else {
+            return new JsonApi(self.im)
+              .post(self.im.config.sassa_api.url + "/api/v1/check_id_number", {
+                data: {
+                  id_number: content,
+                },
+                headers: {
+                  Authorization: ["Token " + self.im.config.sassa_api.token],
+                  "User-Agent": ["Jsbox/SASSA-Registration-USSD"],
+                },
+              })
+              .then(
+                function (response) {
+                  if (!response.data.valid || response.data.underage) {
+                    return error_msg;
+                  } else if (response.data.existing) {
+                    return $(
+                      [
+                        "An application for {{ id_number }} has already been made. The status is: {{ status }}.",
```

Comment:
Where are we expecting to get the status from?

Reply:
So it will just be pending, approved or denied. Pending until we have a answer from SASSA